### PR TITLE
Updates to our design principals document

### PR DESCRIPTION
**Pull request by PaulBone**
Mostly I have added a section on the our biggest design principals.  'Familiarity' was already there, but it has been revised also.

### CODE REVIEW

**Comment by PaulBone** [comment]
Hi @ACleverDisguise, From our conversation yesterday I realised I hadn't written this stuff down.  Is it clear?  You probably don't need to review/validate the principals themselves but I'm also happy to listen to input, particularly if something is surprising.

File: docs/design_principles.txt [hunk 1]
```diff
@@ -16,29 +16,86 @@ Many of these will be described with anti-examples ("don't").  I'd prefer to
...
+We make a number of choices about syntax that will be more familiar to most
+programmers.  For example Functions and blocks use curly-brace syntax of
+C-like languages and the body of a function or block is a series of
+statements.
 
 Likewise we use terminology and names that are going to be more familiar.
 What Haskell calls "Functor" we shall call "Mappable".  We know this
-isn't as accurate as "Functor", but we believe that it being more familiar to
-more people is a greater benefit than this level of accuracy.
+isn't as accurate as "Functor", but we're willing to loose some of that
```
Comment (ACleverDisguise) [inline docs/design_principles.txt:67]: s/loose/lose/

Comment (PaulBone) [reply docs/design_principles.txt:67]: Thanks.

File: docs/design_principles.txt [hunk 2]
```diff
@@ -16,29 +16,86 @@ Many of these will be described with anti-examples ("don't").  I'd prefer to
...
+Sometimes something is familiar to a smaller group of people.  Like ADTs, we
+use the Haskell syntax for ADTs because that's the syntax that's familiar to
+the largest group of people.
 
 Likewise some concepts have no familiar meaning (eg Monad).  We carefully
-weigh whether to include that concept (we will not include GADTs) and it is
-useful enough to include (Monads) we de-emphasise it.
+weigh whether to include that concept at all.  For example monads are very
+useful so we will include but de-emphasise them.  While GADTs are not
+generally useful so we will not include them.
```
Comment (ACleverDisguise) [inline docs/design_principles.txt:80]: I trust you're preparing the bulwarks against the horde of Haskell fanatics who will be here to screech that yes, indeed, GADTs are generally useful, hence the "G" in the name.  :D

I'd suggest instead, "GADTs are more specialist in their use cases so…"

Comment (PaulBone) [reply docs/design_principles.txt:80]: Yeah, I had assumed I would underline "generally" in that sentence when they showed up.  But that I would be unable to convince them of my position regardless of what I said.  In any cause, I like your wording better.

**Comment by ACleverDisguise** [comment]
Last suggestion is more global.  Opening paragraph should have some glurge about "central principals" followed by a list of those three.  The three need punchier titles:

1. Easy Reasoning
2. Familiar Syntax
3. Easy Concurrency and Parallelism

**Comment by PaulBone** [comment]
>     3. Easy Concurrency and Parallelism

I'm not sure that Easy is the right adjective here.  Sure it should be "easy" to manage a large scale program with many different concurrent tasks, but that's not how people think when they read 'easy".  "Fearless" would do but it's been "taken".  I do want "world class" but what if I'm wrong, that's like a radio station saying "we're the best ROCK STATION" but maybe there's a better station for rock some place else.

**Comment by PaulBone** [comment]
Thanks for the feedback @ACleverDisguise.

**Comment by ACleverDisguise** [comment]
Yeah, this looks good.  It's clear and only slightly pretentious.  That's the best approach.

**Pull request merged**